internal/service/alarms/api/openapi.yaml

```diff
@@ -0,0 +1,756 @@
+        '500':
+          description: Internal server error.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+  /O2ims_infrastructureMonitoring/v1/alarms/{alarmEventRecordId}:
+    get:
+      operationId: GetAlarm
+      summary: Retrieve exactly one alarm
+      tags:
+        - alarms
+      parameters:
+        - in: path
+          name: alarmEventRecordId
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AlarmEventRecord'
+        '404':
+          description: The specified AlarmEventRecord was not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+    patch:
+      operationId: AckAlarm
+      summary: Modify exactly one alarm to acknowledge
+      tags:
+        - alarms
+      parameters:
+        - in: path
+          name: alarmEventRecordId
+          required: true
+          schema:
+            type: string
+            format: uuid
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AlarmEventRecordModifications'
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AlarmEventRecordModifications'
+        '404':
+          description: Not found
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Internal server error
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+  /O2ims_infrastructureMonitoring/v1/alarmSubscriptions:
+    get:
+      operationId: GetSubscriptions
+      summary: Retrieve the list of alarm subscriptions
+      tags:
+        - subscriptions
+      parameters:
+        - in: query
+          name: filter
+          schema:
+            type: string
+          allowReserved: true
+          description: Optional filter parameter. Only the attributes in AlarmSubscriptionInfo are filterable.
+          example: (eq,callback,http://my.callback.com)
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/AlarmSubscriptionInfo'
+        '400':
+          description: Bad request
+          content:
```

Comment:
don't see 403 or 404. I think we should be consistent with response codes for list operations

Reply:
added 403. i thought we dont need 404 since we are returning empty list?

Reply:
That was in case you wanted to leave 404 in the previous list